rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java

```diff
@@ -261,24 +263,36 @@ protected void prepareUndoLog(TableRecords beforeImage, TableRecords afterImage)
      * @param rowsIncludingPK the records
      * @return the string
      */
-    protected String buildLockKey(TableRecords rowsIncludingPK) {
+    protected List<LockKey> buildLockKey(TableRecords rowsIncludingPK) {
         if (rowsIncludingPK.size() == 0) {
             return null;
         }
-
-        StringBuilder sb = new StringBuilder();
-        sb.append(rowsIncludingPK.getTableMeta().getTableName());
-        sb.append(":");
-        int filedSequence = 0;
         List<Field> pkRows = rowsIncludingPK.pkRows();
+        List<LockKey> locks = new ArrayList<>();
         for (Field field : pkRows) {
-            sb.append(field.getValue());
-            filedSequence++;
-            if (filedSequence < pkRows.size()) {
-                sb.append(",");
+            LockKey lockKey = new LockKey();
+            lockKey.setTableName(rowsIncludingPK.getTableMeta().getTableName());
+            lockKey.setPk(field.getValue().toString());
+            locks.add(lockKey);
+        }
+        return locks;
+    }
+
+    public String buildLockKeys(List<LockKey> lockKeys) {
```

Comment:
now, the data I think will look like
```
lockKeys : [
    {tableName : a, pk : 1},
    {tableName : a, pk : 2},
    {tableName : a, pk : 3}
]
```
And still have duplicate table name, can you think about a way more simple and more efficient?